src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -26,17 +30,101 @@ export class AwsSamDebugConfigurationProvider implements vscode.DebugConfigurati
         token?: vscode.CancellationToken
     ): Promise<AwsSamDebuggerConfiguration | undefined> {
         if (!AWS_SAM_DEBUG_REQUEST_TYPES.has(debugConfiguration.request)) {
-            vscode.window.showInformationMessage('Invalid request type')
+            vscode.window.showErrorMessage('Invalid request type')
+
+            return undefined
+        }
+
+        if (!AWS_SAM_DEBUG_TARGET_TYPES.has(debugConfiguration.invokeTarget.target)) {
+            vscode.window.showErrorMessage('Invalid invokeTarget.target type')
+
+            return undefined
         }
-        if (
-            debugConfiguration.invokeTarget &&
-            debugConfiguration.invokeTarget.target &&
-            !AWS_SAM_DEBUG_TARGET_TYPES.has(debugConfiguration.invokeTarget.target)
-        ) {
-            vscode.window.showInformationMessage('Invalid invokeTarget.target type')
+
+        if (debugConfiguration.invokeTarget.target === 'template') {
+            if (!isTemplateDebugConfigValid(debugConfiguration)) {
+                return undefined
+            }
+        } else if (debugConfiguration.invokeTarget.target === 'code') {
+            if (
+                !debugConfiguration.lambda ||
+                !debugConfiguration.lambda.runtime ||
+                !samLambdaRuntimes.has(debugConfiguration.lambda.runtime)
+            ) {
+                vscode.window.showErrorMessage(
+                    'Debug config of invokeTarget.target: code must have a valid lambda.runtime'
+                )
+
+                return undefined
+            }
         }
+
         vscode.window.showInformationMessage('Not implemented')
 
-        return undefined
+        return debugConfiguration
+    }
+}
+
+function isTemplateDebugConfigValid(debugConfiguration: AwsSamDebuggerConfiguration): boolean {
```

Comment:
The common pattern for validating is to have it return a validation message (string), and undefined when there are no validation issues. This centralizes the actions taken (such as notifying the user).

It's acceptable to separate warning checks from error checks out to separate methods